src/components/Suspense.astro

```diff
@@ -1,7 +1,72 @@
 ---
-const idx = Astro.locals.suspend(Astro.slots.render("default"));
+type PendingThenable<T> = { status: "pending" };
+type FulfilledThenable<T> = { status: "fulfilled"; value: T };
+type RejectedThenable<T> = { status: "rejected"; reason: unknown };
+
+type Thenable<T> = Promise<T> &
+  (PendingThenable<T> | FulfilledThenable<T> | RejectedThenable<T>);
+
+function instrument<T>(promise: Promise<T>): Thenable<T> {
+  const thenable = promise as Promise<T> & PendingThenable<T>;
+  thenable.status = "pending";
+  thenable.then(
+    (value) => {
+      const fulfilled = promise as Promise<T> & FulfilledThenable<T>;
+      fulfilled.status = "fulfilled";
+      fulfilled.value = value;
+    },
+    (error) => {
+      const rejected = promise as Promise<T> & RejectedThenable<T>;
+      rejected.status = "rejected";
+      rejected.reason = error;
+    }
+  );
+  return thenable;
+}
+
+function sleep(ms: number) {
```

Comment:
Can we move this to a separate `utils.ts` file in the `simple-stack-stream` package? Have a larger plan to create a `simple-stack-utils` package with shared utilities like this

Reply:
yeah for sure, i was just being lazy hehe